File: binaryplus/src/SoundSystem.cpp
#include "dllimport.hpp"
#include <unicode_conversion.hpp>
#include "SoundSystem.hpp"

using namespace std;
using namespace cppimp;
using namespace uniconv;

namespace cpp {
    void SoundSystem::PlaySound(u16string filepath, bool wait) {
        auto filepathPtr = U16StringToUnicode(filepath);

        SoundSystem_PlaySound(filepathPtr, wait);
    }
}

File: binarysharp/SoundSystem.cs
using System.Runtime.InteropServices;
using Utility;
using Cpp;

namespace Cs {
    public class SoundSystem {
        public static void PlaySound(string filepath, bool wait = false)
        {
            CsImp.SoundSystem.PlaySound(TypeConvert.StringToPtr(filepath),wait);
        }
    }
}